### DISCUSSION TITLE
Identify and switch layer

### DISCUSSION THREAD

**Discussion by igorline** [manna-harbour/miryoku#176] [Q&A] [answered]
Hey! I'm using dvorak layout to typing in English, but I need to switch to qwerty to type in another language. I want to implement switch lang logic and set it on unused key on a corne keyboard. I don't understand how can I identify current layer and how to switch to a different one using miryoku.
Previously with my qmk keymap I did this:

```
void switch_language(void) {
    switch (biton32(default_layer_state)) {
        case _DVORAK:
          set_single_persistent_default_layer(_QWERTY); 
          break;
        default:
          set_single_persistent_default_layer(_DVORAK);
          break;
    }
    tap_code16(LSFT(LGUI(KC_F2)));
}
```

Second keycode tap would trigger my os to switch language

Thanks in advance!

**Comment by manna-harbour** [comment, marked as answer]
There's a built-in Extra layer, which is QWERTY by default. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features.

**Comment by manna-harbour** [comment]
The layer name enums for Base and Extra are `U_BASE` and `U_EXTRA`. If you add your code to `manna-harbour_miryoku.c` you can use those names for the layers. If you add your code to another file you can use the layer numbers instead, `0` and `1` respectively.
